### PR TITLE
opentelemetry instrumentation initial metrics in MQTT broker

### PR DESCRIPTION
**Pull request by nlcamp**
Added instrumentation to MQTT broker for a single metric:  mqtt.broker.client.messages.received. This metric is intended to represent the total number of client messages received by the broker.

Testing needs to be done to confirm that performance of the broker is not impacted negatively.

### CODE REVIEW

**Comment by vadim-kovalyov** [comment]
> Added instrumentation to MQTT broker for a single metric: mqtt.broker.client.messages.received. This metric is intended to represent the total number of client messages received by the broker.
> 
> Testing needs to be done to confirm that performance of the broker is not impacted negatively.

Noel, before the review, is there a high level design or documentation for broker metrics? It would be helpful for us (broker v-team) to understand the design/north star solution to see the whole picture.

File: mqtt/mqtt-otel/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        .unwrap();
+
+    Ok(response)
+}
+
+pub fn init_prometheus_metrics_exporter() -> Result<PrometheusExporter, MetricsError> {
+    opentelemetry_prometheus::exporter().try_init()
+}
+
+pub fn create_prometheus_server(
```
Comment (varunpuranik) [inline mqtt/mqtt-otel/src/lib.rs:42]: This will not integrate very well with the EdgeHub container. We need to discuss how to do that..

**Review by varunpuranik — COMMENTED** [review]
Can we have a design for the MQTT Broker metrics and how they will be added? We also need to make sure they integrate well with the existing EdgeHub metrics.

**Comment by nlcamp** [comment]
> > Added instrumentation to MQTT broker for a single metric: mqtt.broker.client.messages.received. This metric is intended to represent the total number of client messages received by the broker.
> > Testing needs to be done to confirm that performance of the broker is not impacted negatively.
> 
> @nlcamp, before the review, is there a high level design or documentation for broker metrics? It would be helpful for us (broker v-team) to understand the design/north star solution to see the whole picture.

@vadim-kovalyov We skipped formal documentation so we could quickly demonstrate a single metric flowing end to end using open-telemetry APIs. However now that we have that, it does seem like a good idea to pause and provide some high level design documentation. We'll start with high level block diagrams and sequence diagrams and go from there. I'll put the PR back into Draft status until this is completed.

**Comment by nlcamp** [comment]
> Can we have a design for the MQTT Broker metrics and how they will be added? We also need to make sure they integrate well with the existing EdgeHub metrics.

Will do. See reply to @vadim-kovalyov 's comment [here](https://github.com/Azure/iotedge/pull/4487#issuecomment-793042135)

File: mqtt/mqtt-otel/src/lib.rs (hunk 1 above)
Comment (nlcamp) [reply mqtt/mqtt-otel/src/lib.rs:42]: Will include mention/depiction of this in the design docs/diagrams.  These docs/diagrams should help with the discussion about the path forward integrating with the EdgeHub container.

**Comment by nlcamp** [comment]
Closing PR until larger observability design process is completed.

**Pull request closed without merging**